app/models/file_upload_handler.rb

```diff
@@ -11,7 +11,8 @@ class FileUploadHandler
   # key      - The key of the ScenarioAttachment, e.g. 'esdl_file'
   # scenario - The scenario the file should be attached to
   def initialize(file, filename, key, scenario)
-    @file = file
+    @file = Tempfile.new(filename)
```

Comment:
I'm unclear why the uploaded contents are written to a tempfile. Was this because of my suggestion for scanning the first 1k of ESDL files?

I mistakenly thought the `file` was already a tempfile, but now I see it's a string. Could this be reverted with the validation doing...

```ruby
errors.add('...') unless @file[0...1024].include?('<esdl:EnergySystem')
```

... or am I missing something?

---

I was tripped by by `file` being a string, because I was expecting an [ActionDispatch::Http::UploadedFile](https://api.rubyonrails.org/v6.0.3.3/classes/ActionDispatch/Http/UploadedFile.html), which is what would normally result from an uploaded file. Instead it appears as though the file contents are provided to ETEngine as a string rather than using multipart/form-data. I'm a bit split on this:

* It's not consistent with curve uploads which use multipart/form-data, and therefore may be confusing to API users.
* I'm unclear whether there could be encoding issues providing the file as a string rather than a binary upload. I'm guessing not, but... 🤷‍♂️
* This would be an issue if you ever wanted someone to upload an image, or some other form of binary/encoded data.
* It wouldn't be possible to create an HTML form to add a file directly; we'd need a proxy which received the upload, read the contents into a string, then submitted it to ETEngine. Though to be fair, it's not clear if we'd ever need this.

Would this be a real pain to change in the etm-esdl app? If so, I'm fine leaving it for now, but it feels awkward to design the ETEngine API around etm-esdl, rather than the other way around.

Sorry to keep coming back to you with more. 😟

Reply:
No worries, that's how you learn 😄 I'm rushing this a bit, because I was hoping it could join this deploy, but maybe I need to slow down because I'm starting to get sloppy... Sorry about that!

Yep, I saw your earlier suggestion and I thought: why did I ever choose to make this a String upload? So I already went back to the etm-esdl app and tried to change the API to send form-data to etengine, because I agree that all the upload API's should be similar. However, the ESDL file is changed by the app before it gets sent to ETEngine and after spending 20min trying to convince Flask to put it back into form-data, I thought 🤷 So here we see the result of that haha.

So again - excuse my sloppiness. I'll slow down and go make it pretty!